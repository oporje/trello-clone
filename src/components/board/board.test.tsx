import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Enzyme from 'enzyme';
import Board from './board';
import List from '../list/list';
import React from 'react';
import { shallow } from 'enzyme';


Enzyme.configure({ adapter: new Adapter() });

test('renders the Board component', () => {
  const boardProps = {
    boardId: `12345`,
    name: 'test'
  };
  const component = shallow(<Board board={boardProps} />);
  expect(component).toMatchSnapshot();
});

test('renders the Board with props', () => {
  const boardProps = {
    boardId: `12345`,
    name: 'test'
  };
  const component = shallow(<Board board={boardProps} />);
  expect(component).toMatchSnapshot();
  expect(component.find(List).length).toEqual(0);
  const addListBtn = component.find('button.add-list-btn');
  addListBtn.simulate('click');
  expect(component.find(List).length).toEqual(1);
});
