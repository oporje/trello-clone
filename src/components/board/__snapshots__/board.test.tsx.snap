// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Board component 1`] = `ShallowWrapper {}`;

exports[`renders the Board with props 1`] = `ShallowWrapper {}`;
